### DISCUSSION TITLE
Lung Nodule Detection Training Parameters

### DISCUSSION THREAD

**Discussion by junxiant** [Project-MONAI/model-zoo#600] [Q&A]
Hello, I tried training the lung nodule detection following the tutorial and it worked out well so far, but a few questions,

1. any reason for score_thresh of 0.02 and nms_thresh of 0.22? 
2. in set_box_selector_parameters, does it use class agnostic nms? 

I am attempting to train a multi class nodule detection, would i need to increase the score_thresh if nms is class agnostic?

thank you

**Comment by hari3100** [comment]
Hi can you share the tutorial you watched? , I'm trying to use the same sort of model , but for CT scan images

**Reply by junxiant** [reply]
Hi, i didnt watch any tutorial. 

1. I played around with the score_thresh and nms_thresh, modifying it has impact on the results.
2. I'm not sure about this, but i looked through the source code and i think it is class agnostic.

There are some other discussions which helped so i will link them here:
https://github.com/Project-MONAI/model-zoo/issues/578#issuecomment-2237171911
https://github.com/Project-MONAI/MONAI/discussions/7917

**Reply by hari3100** [reply]
Hi,
Thank you for such a quick response, 
Im not exactly trying to re-train the model or anything but rather just use the inference part of it in my python workflow, where at the end of it , i could upload a nifti file of a abnormal lung , the model could predict/ detect some nodule and show me that nodule , like some mask or an outline of that detected nodule, maybe that output could be a nifti file too . 

I'm trying to create a python inference of this model. And Im facing a lot of issue.
Right from how to parse the config files , as when I tried it , the inferer file is giving me some issues .
And im even not sure how to proceed with the actual prediction part in python code.
I'm new to this model , i have used whole body ct segmentation model and it also had some similar looking config files , so I tried doing the same steps atleast till parsing the config file , but i couldn't do that fully either.

Im exactly not sure how the whole process should even be carried on or coded.
Do I need to have luna data to do the inference?....DO I need to have some other 3rd party files before running the inference, other than the files that I downloaded from monai while downloading the model?....
Im using lung CT scans , as dicom series in a folder.

Any help related to this or discussing and giving me some clarity would be greatly appreciated, My doubts can be very naive, thank you for your patience.

Thank you @junxiant .
**I have uploaded the screenshot of what I have done till now and the files i have  , which most probably is a big pile of poo🙆‍♂️**
![image](https://github.com/user-attachments/assets/c3050adc-6ac3-4541-9274-d7c4f8f7bb08)
![Screenshot 2024-08-21 152505](https://github.com/user-attachments/assets/c0682b94-a2b7-445c-8524-b923a9fdbb6d)
![Screenshot 2024-08-21 152610](https://github.com/user-attachments/assets/a53021de-d114-48e1-b5c6-9eee90f5633a)
![Screenshot 2024-08-21 152638](https://github.com/user-attachments/assets/cb8aa362-f384-4790-8410-303b83f3781b)
![Screenshot 2024-08-21 152707](https://github.com/user-attachments/assets/35512ed3-04eb-44ed-b366-87f9f84fe9a5)
![Screenshot 2024-08-21 152729](https://github.com/user-attachments/assets/36aee2aa-5480-4ee3-af28-94a389792788)

**Comment by junxiant** [comment]
Hi, if i am understanding you correctly, you want to:

1. Use the pretrained model as it is to do inference
2. Input would be in nifti format, outputs will also be in nifti format

I think you can do it in two ways, if we look at this link guide https://github.com/Project-MONAI/model-zoo/tree/dev/models/lung_nodule_ct_detection 
![image](https://github.com/user-attachments/assets/449659d5-8781-40ad-9401-e8de7a4f0f70)
We see there's inference code to be used, but this will require you to modify the dataset json to point towards your own raw input data. 

So ultimately i think you need to 
a. put raw files inside a folder
b. have a json file that contains the directory to the raw input data

But i dont think this is feasible? or maybe not straightforward enough. the outputs... i dont remember where it would be stored.

The other way is to use monailabel to start a server where you can either use it as an API or with a 3rd party viewer such as 3D slicer (https://www.slicer.org/).

This way, you need to:
1. prep your files inside the folder and have the --studies argument point to the directory, see the command below as an example. The input data i have is nifti (.nii.gz). 
2. start the monailabel server
3. use 3D Slicer with the MONAILabel extension
4. once you ran inference with the input data, outputs will be displayed on the viewer, you can save the outputs (annotations/labels) and it will be saved in the same datasets/Task06_Lung/.. folder

```
monailabel start_server --app apps/monaibundle --studies datasets/Task06_Lung/imagesTs --conf models lung_nodule_ct_detection
```

or you can 
1. start the monailabel server
2. use the monailabel server's API as it is
3. you will need to call the inference function with your input data
